### PR TITLE
Match list

### PR DESCRIPTION
**Pull request by YoniKiriaty**

### CODE REVIEW

File: vite-project/src/scouter/MatchList.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    setMatches(filtered);
+    navigate("/");
+  }
+
+  return (
+    <div className="match-list">
+      {matches.map((match, index) => (
+        <Collapsible
+          trigger={`${matchName} ${match[matchName]}`}
+          triggerStyle={{ borderStyle: "solid" }}
```
Comment (Yotam7) [inline vite-project/src/scouter/MatchList.tsx:43]: consider moving css to separate css file

File: vite-project/src/scouter/ScouterQuery.tsx [hunk 2]
```diff
@@ -9,54 +9,39 @@ interface ScouterQueryProps {
...
+        return <CounterQuery name={name} />;
       case "checkbox":
-        return (
-          <CheckboxQuery
-            name={queryName}
-            required={required || false}
-            initialValue={initialValue}
-          />
-        );
+        return <CheckboxQuery name={name} required={required || false} />;
```
Comment (Yotam7) [inline vite-project/src/scouter/ScouterQuery.tsx:25]: here it is required || false
↓↓↓

File: vite-project/src/scouter/ScouterQuery.tsx [hunk 3]
```diff
@@ -9,54 +9,39 @@ interface ScouterQueryProps {
...
       case "list":
         return (
-          <ListQuery
-            name={queryName}
-            required={required}
-            initialValue={initialValue}
-            list={list ? list : []}
-          />
+          <ListQuery name={name} required={required} list={list ? list : []} />
```
Comment (Yotam7) [inline vite-project/src/scouter/ScouterQuery.tsx:28]: ^^^
here it is required
↓↓↓

File: vite-project/src/scouter/ScouterQuery.tsx [hunk 4]
```diff
@@ -9,54 +9,39 @@ interface ScouterQueryProps {
...
         );
       case "radio":
         return (
-          <RadioQuery
-            name={queryName}
-            required={required}
-            initialValue={initialValue}
-            list={list ? list : []}
-          />
+          <RadioQuery name={name} required={required} list={list ? list : []} />
```
Comment (Yotam7) [inline vite-project/src/scouter/ScouterQuery.tsx:32]: ^^^
and here it is also required
think about weather "|| false" is required, if its not remove it, if it is, add it where needed

File: vite-project/src/scouter/ScoutingTab.tsx [hunk 5]
```diff
@@ -15,32 +14,31 @@ function ScouterApp() {
...
+    navigate("/", { state: formValues });
   }
 
   function clearQueryStorage() {
     Object.keys(localStorage)
-      .filter((item) => item.startsWith("Queries/"))
-      .forEach((item) => localStorage.removeItem(item));
+      .filter((item) => item.startsWith(localStorageTabName))
+      .forEach((item) => {
+        console.log(item);
```
Comment (Yotam7) [inline vite-project/src/scouter/ScoutingTab.tsx:25]: consider removing console logging

File: vite-project/src/scouter/ScoutingTab.tsx [hunk 6]
```diff
@@ -50,15 +48,18 @@ function ScouterApp() {
       <MapQuery
         name="CRESCENDO"
         width={640}
```
Comment (Yotam7) [inline vite-project/src/scouter/ScoutingTab.tsx:50]: consider moving values to const
(also next line)

File: vite-project/src/scouter/querytypes/RadioQuery.tsx [hunk 7]
```diff
@@ -1,28 +1,24 @@
...
         <input
           type="radio"
           id={item}
           name={name}
           value={item}
           required={required}
-          onChange={() => localStorage.setItem(name, item)}
+          onChange={() => localStorage.setItem(localStorageKey, item)}
           defaultChecked
         />
```
Comment (Yotam7) [inline vite-project/src/scouter/querytypes/RadioQuery.tsx:23]: having such large item inside ternary operator is hard to read, consider moving to separate variable

File: vite-project/src/scouter/ScoutingTab.tsx (hunk 6 above)
Comment (YoniKiriaty) [reply vite-project/src/scouter/ScoutingTab.tsx:50]: this file is where you config everything

File: vite-project/src/scouter/MatchList.tsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+  if (latestMatch?.[matchName]) {
+    matches.push(latestMatch);
+    localStorage.setItem(
+      matchesTab + latestMatch[matchName],
+      JSON.stringify(latestMatch)
+    );
+  }
+
+  function removeMatch(qualNumber: string, index: number) {
```
[the remark this answers is not in the document]
Comment (YoniKiriaty) [reply vite-project/src/scouter/MatchList.tsx:30]: the form and QR recognize inputs as a string, so the qual number is a string

**Review by Yotam7 — APPROVED** [review]

**Pull request merged**